app/components/Login.react.js

```diff
@@ -52,25 +54,62 @@ class Login extends Component {
         this.buildOauthUrl = this.buildOauthUrl.bind(this);
         this.oauthPopUp = this.oauthPopUp.bind(this);
         this.logIn = this.logIn.bind(this);
-        this.checkAnyUserAccountIsSaved = this.checkAnyUserAccountIsSaved.bind(this);
     }
 
-    componentWillReceiveProps(nextProps) {
-        this.checkAnyUserAccountIsSaved().then(isSaved => {
-            if (isSaved) {
-                window.location.assign(connectorUrl);
-            }
-        });
+    componentDidMount() {
+        const {serverType} = this.state;
+        const that = this;
+        const interval = setInterval(() => {
+              usernameLogged = cookie.load('db-connector-user');
+              if (usernameLogged) {
+                  if (serverType === ONPREM) {
+                      that.setState({
+                          status: 'authorized',
+                          statusMessage: 'Saving user information...'
+                      });
+                      that.saveDomainToSettings().then(res => {
+                          if ( res.status === 200 ) {
+                              window.location.assign(connectorUrl);
+                          } else {
+                              that.setState(
+                                  status: 'failure',
+                                  statusMessage: 'An error occured. Please try again.'
```

Comment:
🐮  spelling: `occurred`

Reply:
it would be helpful to display the error message from `res.text()` too, especially when we're helping customers out

Reply:
See https://github.com/plotly/plotly-database-connector/pull/180/files#diff-fd9a17590082224a502b26de40234d29L103 for an example of this error handling flow